pkg/servicemanager/startuprecorder/startuprecorder.go

```diff
@@ -0,0 +1,71 @@
+package startuprecorder
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+type ServiceData struct {
+	Name         string        `json:"name"`
+	Dependencies []string      `json:"dependencies"`
+	Start        time.Time     `json:"start"`
+	Ready        time.Time     `json:"ready"`
+	TimeToReady  time.Duration `json:"timeToReady"`
+}
+
+type StartupRecorder struct {
+	Services              []ServiceData `json:"services"`
+	MicroshiftStart       time.Time     `json:"microshiftStart"`
+	MicroshiftReady       time.Time     `json:"microshiftReady"`
+	MicroshiftTimeToReady time.Duration `json:"microshiftTimeToReady"`
+
+	m sync.Mutex
+}
+
+func New() *StartupRecorder {
+	return &StartupRecorder{}
+}
+
+func (l *StartupRecorder) LogServiceReady(serviceName string, dependencies []string, start time.Time, ready time.Time) {
+	serviceData := ServiceData{
+		Name:         serviceName,
+		Dependencies: dependencies,
+		Start:        start,
+		Ready:        ready,
+		TimeToReady:  ready.Sub(start),
+	}
+
+	l.m.Lock()
+	l.Services = append(l.Services, serviceData)
+	l.m.Unlock()
```

Comment:
You can use defer here. It's a simple flow, so it'll always unlock, but `defer` is often used with `Unlock()`, `Close()` and similar.
If you're worried about `klog.InfoS` being covered by mutex, you can move the log above `Lock()`
```suggestion
	l.m.Lock()
    defer l.m.Unlock()
	l.Services = append(l.Services, serviceData)
```

Reply:
done